aspnetcore/signalr/configuration.md

```diff
@@ -0,0 +1,202 @@
+---
+title: ASP.NET Core SignalR configuration
+author: rachelappel
+description: Configure ASP.NET Core SignalR Apps
+manager: wpickett
+monikerRange: '>= aspnetcore-2.1'
+ms.author: rachelap
+ms.date: 06/30/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: signalr/configuration
+---
+
+# ASP.NET Core SignalR configuration
+
+## JSON/MessagePack Serialization Options
+
+ASP.NET Core SignalR supports two protocols for encoding messages, JSON and MessagePack. Each protocol has configuration options that can be used to configure serialization.
+JSON Serialization Options
+
+JSON serialization can be configured on both the Server and .NET Client by providing a delegate to the AddJsonHubProtocol method. The options object received by that delegate has a PayloadSerializedSettings which is a JSON.NET JsonSerializerSettings object that can be used to configure serialization of arguments and return values
+
+On the server, add `AddJsonHubProtocol` to the `AddSignalR` call in `ConfigureServices`. Use PascalCase property names instead of the default camelCase names. See the [JSON.NET Documentation](https://www.newtonsoft.com/json/help/html/Introduction.htm) for more details.
+
+```csharp
+    services.AddSignalR()
+        .AddJsonHubProtocol(options => {
+            options.PayloadSerializerSettings.ContractResolver = 
+            new DefaultContractResolver();
+        });
+}
+```
+
+On a .NET client, the same `AddJsonHubProtocol` extension method exists on `HubConnectionBuilder`. The `Microsoft.Extensions.DependencyInjection` namespace must be imported to see the method:
+
+```csharp
+// At the top of the file:
+using Microsoft.Extensions.DependencyInjection;
+
+// When constructing your connection:
+var connection = new HubConnectionBuilder();
+    .AddJsonHubProtocol(options => {
+        options.PayloadSerializerSettings.ContractResolver = 
+            new DefaultContractResolver();
+    });
+```
+
+### MessagePack Serialization Options
+
+MessagePack serialization can be configured by providing a delegate to the `AddMessagePackProtocol` call. See [MessagePack in SignalR](xref:signalr/messagepack) for more details.
+
+> [!NOTE]
+> It's not possible to configure JSON or MessagePack serialization in the JavaScript client at this time.
+
+### Configure [HubOptions](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.signalr.huboptions) for SignalR
```

Comment:
I know we don't have "xref" for these, but if we're going to link to APIs, then "/dotnet/api/..." is probably better. Less hardcoded and preserves language